addons/binding/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/internal/discovery/HomeAssistantDiscovery.java

```diff
@@ -161,7 +161,7 @@ public void receivedMessage(ThingUID connectionBridge, MqttBrokerConnection conn
 
         Map<String, Object> properties = new HashMap<>();
         HandlerConfiguration handlerConfig = topicParts.toHandlerConfiguration();
-        handlerConfig.toProperties(properties);
+        handlerConfig.appendToProperties(properties);
```

Comment:
Make it like so: `properties = handlerConfig.appendToProperties(properties);`
I'm not sure if the java compiler is smart enough to figure out that it means the same (in c++ that's possible). But for someone reading the code it is more obvious.

Reply:
yes, should have thought about this...
Its too late after a working day...